src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -99,31 +128,23 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
     } else if (processSingle) {
       this.launch()
     } else if (type === EntitySelectionType.processAll) {
-      this.setState({ message: 'Fetching data...' })
-      const entities = _.map('name', await Workspaces.workspace(namespace, name).entitiesOfType(rootEntityType))
       this.createSetAndLaunch(entities)
     } else if (type === EntitySelectionType.chooseRows) {
-      const entities = _.keys(selectedEntities)
-      if (_.size(entities) === 1) {
-        this.launch(rootEntityType, _.head(entities))
+      if (entities.length === 1) {
+        this.launch(rootEntityType, entities[0])
       } else {
         this.createSetAndLaunch(entities)
       }
     } else if (type === EntitySelectionType.processMergedSet) {
-      if (_.size(selectedEntities) === 1) {
-        const { entityType, name } = _.values(selectedEntities)[0]
+      if (entities.length === 1) {
```

Comment:
I think we would want that case to take the first branch; this is running against the entities of the set, not the set itself, the alternative would be running against the members of a set with 1 member.

Reply:
We could, but it would be a change in behavior, and also doesn't match the message we give, e.g. 'Selected rows will have their membership combined...'

Reply:
Hmm, that's true. Fair enough.